### ISSUE TITLE
vmtoolsd process eat 100% CPU for 5-6 second when BIRD runned.

### ISSUE THREAD

**Issue by takenek** [vmware/open-vm-tools#186]
Hello
I create little bgp router on Linux Debian 9 and BIRD.
I use Open-VM-Tools version: 
ii  open-vm-tools                  2:10.1.5-5055683-4             amd64        Open VMware Tools for virtual machines hosted on VMware (CLI)

and BIRD version:
ii  bird                           1.6.3-2                        amd64        Internet Routing Daemon

I use only IPv6 routing and got strange problem.
Every 10 second vmtoolsd process eat 100% CPU for 5-6 second all the time.

Looks like vmtoolsd got problem when coming big import to kernel routing table.

Best Regards
Marek Krolikowski

**Comment by oliverkurth** [comment]
How large is the routing table? Does it change every time? If possible, attach a sample of the routing table.

**Comment by takenek** [comment]
Hello,
root@pl-bgp:~# ip -6 r s |wc -l
43640

root@pl-bgp:~# ip -4 r s |wc -l
2

root@pl-bgp:~# ip -6 r s
2a00:4800:2e0::/44 via 2001:470:12:58::1 dev bgp-henet proto bird metric 1024  pref medium
2a0b:5b00::/29 via 2a06:1287:330e:3::1 dev bgp-lax proto bird metric 1024  pref medium
2a0b:a380:1::/64 via 2a06:1287:330e:a::2 dev bgp-sea proto bird metric 1024  pref medium
2c0f:f6e0::/32 via 2a06:1287:330e:1::1 dev bgp-nl proto bird metric 1024  pref medium
2c0f:f7a8:8211::/48 via 2a06:1287:330e:16::2 dev bgp-fr proto bird metric 1024  pref medium

root@pl-bgp:~# cat /etc/bird/bird6.conf |grep time
scan time 10;


Best Regards
Marek Krolikowski

**Comment by oliverkurth** [comment]
Thank you. This a very large routing table, and this is most likey the cause for the CPU load. I filed an internal bug for this. To workaround, you can configure the poll interval in tools.conf, which is 30 seconds by default. For example, to set it to 15 minutes, do:

```
[guestinfo]
poll-interval=900
```

You can also disable it completely by setting the value to 0. Beware though that in that case no guest information will be passed to the host.

**Comment by takenek** [comment]
Hello,
Yes this is full IPv6 routing table, but when i disable open-vm-tools my load average drop down to 0.01 and CPU/Server boring, that`s why i am sure this is a problem with open-vm-tools and support big routing table.

I just check and after put:
[guestinfo]
poll-interval=900
to my tools.conf it`s little better, because pull interval change to 900 sec so my CPU back to almost normal.

Best Regards
Marek Krolikowski

**Comment by gravasio** [comment]
Same problem here...
I'm loading the BGP full routing table for both IPv4 and IPv6.

**Comment by takenek** [comment]
Put:
[guestinfo]
poll-interval=900
And it`s better :-)

**Comment by gravasio** [comment]
Yep I've already put the lines, It's better but still annoying ;-P
I also tried the vmware shipped tools and they have an almost identical issue (60% and not 100% CPU)

**Comment by brandoncazander** [comment]
@oliverkurth will this issue be updated when your internal issue is resolved?

**Comment by oliverkurth** [comment]
Yes. Most likely the fix will first appear in the 'devel' branch.

**Comment by takenek** [comment]
When You release patch for this?
I use Debian version:
ii  open-vm-tools                  2:10.1.5-5055683-4+deb9u1      amd64        Open VMware Tools for virtual machines hosted on VMware (CLI)

Best Regards
Marek Krolikowski

**Comment by andamasov** [comment]
Happy new year!
This still happens with anything that has many routes

**Comment by csalcedo11** [comment]
Same here... Any update on this? We have had to turn polling off in order to workaround the bug.

**Comment by MrXermon** [comment]
The error still persists. In my eyes it's not that useful to push the whole routing-table from a router vm to the hypervisor. It'd be cool to have a option to just disable the publication of the routing-table to the hypervisor. Currently i've just disabled the whole polling by setting the variable mentioned above to 0.

**Comment by lemke1458** [comment]
You can specify the max number of routes to gather in the latest code.

See https://github.com/vmware/open-vm-tools/issues/351

**Comment by MrXermon** [comment]
@lemke1458 Is there a option for IPv4, too? Because i don't think it's a IPv6 specific problem.

**Comment by lemke1458** [comment]
Yes.  "max-ipv4-routes".

See https://github.com/vmware/open-vm-tools/blob/fbfe0cbc116b6a171c29f29560d6dd1891a0870d/open-vm-tools/lib/include/conf.h#L146

IIRC, IPv6 was the big time sink because of how expensive it is to read the info from /proc.  IPv4 was far quicker.

**Comment by MrXermon** [comment]
@lemke1458 Cool! Thank's for the update! :)
